src/Translator.php

```diff
@@ -0,0 +1,542 @@
+
+    /**
+     * Set mtime to a year ago so that transifex will see these as obsolete
+     */
+    private function setJsonAndYmlFileTimes()
+    {
+        $date = date('YmdHi.s', strtotime('-1 year'));
+        foreach ($this->modulePaths as $modulePath) {
+            $name = $this->getModuleName($modulePath);
+            $this->log("Pulling sources from transifex for <info>{$name}</info> (min %{$this->txMinimumPerc} delta)");
+            $ymlLang = $this->getLangDirectory($modulePath);
+            if ($ymlLang) {
+                $this->exec("find $ymlLang -type f \( -name \"*.yml\" \) -exec touch -t $date {} \;");
+            }
+            foreach ($this->getJSLangDirectories($modulePath) as $jsLangDir) {
+                $this->exec("find $jsLangDir -type f \( -name \"*.json*\" \) -exec touch -t $date {} \;");
+            }
+        }
+    }
+
+    /**
+     * Update sources from transifex
+     */
+    private function transifexPullSource()
+    {
+        foreach ($this->modulePaths as $modulePath) {
+            // ensure .tx/config is up to date
+            $contents = file_get_contents("$modulePath/.tx/config");
+            if (strpos($contents, '[o:') === false) {
+                $this->exec('tx migrate', $modulePath);
+                // delete .bak files created as part of tx migrate
+                foreach (scandir("$modulePath/.tx") as $filename) {
+                    if (pathinfo($filename, PATHINFO_EXTENSION) !== 'bak') {
+                        continue;
+                    }
+                    $this->log("Deleting $modulePath/.tx/$filename");
+                    unlink("$modulePath/.tx/$filename");
+                }
+            }
+            // pull from transifex
+            $this->exec("tx pull -a -s -t -f --minimum-perc={$this->txMinimumPerc}", $modulePath);
+        }
+    }
+
+    /**
+     * Merge any missing keys from old yaml content into yaml files
+     */
+    private function mergeYaml(): void
+    {
+        $this->log('Merging local yaml files');
+        foreach ($this->originalYaml as $path => $contentYaml) {
+            // If there are any keys in the original yaml that are missing now, add them back in.
+            $rawYaml = file_get_contents($path);
+            $parsedYaml = Yaml::parse($rawYaml);
+            $contentYaml = $this->arrayMergeRecursive($contentYaml, $parsedYaml);
+            // Write back to local
+            file_put_contents($path, Yaml::dump($contentYaml));
+        }
+        $this->log('Finished merging ' . count($this->originalYaml) . ' yaml files');
+    }
+
+    /**
+     * Tidy yaml files using symfony yaml
+     */
+    private function cleanYaml()
+    {
+        foreach ($this->modulePaths as $modulePath) {
+            $name = $this->getModuleName($modulePath);
+            $this->log("Cleaning YAML sources for <info>{$name}</info>");
+            $num = 0;
+            foreach (glob($this->getLangDirectory($modulePath) . "/*.yml") as $sourceFile) {
```

Comment:
```suggestion
            foreach (glob($this->getLangDirectory($modulePath) . '/*.yml') as $sourceFile) {
```
This one's a bit of a nitpick, I won't hold up the PR for it but it makes sense to flag since there are other changes that do need to be made.

Reply:
Updated